intake_geopandas/regionmask.py

```diff
@@ -0,0 +1,78 @@
+from intake.source.base import Schema
+
+from .geopandas import GeoPandasFileSource
+
+
+class RegionmaskSource(GeoPandasFileSource):
+    name = 'regionmask'
+
+    def __init__(
+        self,
+        urlpath,
+        use_fsspec=False,
+        storage_options=None,
+        bbox=None,
+        geopandas_kwargs=None,
+        metadata=None,
+        regionmask_kwargs=None,
+    ):
+        """
+        urlpath : str or iterable, location of data
+            Either the absolute or relative path to the file or URL to be
+            opened. Some examples:
+            - ``{{ CATALOG_DIR }}data/states.shp``
+            - ``http://some.domain.com/data/states.geo.json``
+        use_fsspec: bool
+            Whether to use fsspec to open `urlpath`. By default, `urlpath` is passed
+            directly to GeoPandas, which opens the file using `fiona`. However, for some
+            use cases it may be beneficial to read the file using `fsspec` before
+            passing the resulting bytes to GeoPandas (e.g., when using `fsspec`
+            caching).
+            Note that fiona/GDAL and `fsspec` have mutually-incompatible URL chaining
+            syntaxes, so the URLs passed to each may be significantly different.
+        storage_options: dict
+            Storage options to pass to fsspec when opening. Only used when
+            `use_fsspec=True`.
+        bbox : tuple | GeoDataFrame or GeoSeries, default None
+            Filter features by given bounding box, GeoSeries, or GeoDataFrame.
+            CRS mis-matches are resolved if given a GeoSeries or GeoDataFrame.
+        geopandas_kwargs : dict
+            Any further arguments to pass to geopandas's read_file function.
+        regionmask_kwargs : dict
+            Any further arguments to pass to regionmask.from_geopandas.
+        """
+        self._regionmask_kwargs = regionmask_kwargs or {}
+
+        super().__init__(
+            urlpath=urlpath,
+            metadata=metadata,
+            use_fsspec=use_fsspec,
+            storage_options=storage_options,
+            geopandas_kwargs=geopandas_kwargs,
+            bbox=bbox,
+        )
+
+    def _open_dataset(self):
+        try:
+            import regionmask
+        except ImportError:
+            raise ImportError('regionmask must be installed')
+        super()._open_dataset()
+        self._dtypes = self._dataframe.dtypes.to_dict()
+        self._dtypes = {n: str(t) for (n, t) in self._dtypes.items()}
+        self._dataframe = regionmask.from_geopandas(
+            self._dataframe, **self._regionmask_kwargs
+        )
+
+    def _get_schema(self):
```

Comment:
As far as I can see, this method is identical to the one in the base class, so it shouldn't need to be overridden.

Reply:
the dtypes is different. I want to pass the `dtypes` from the `dataframe` and not from the `regionmask.region` object. didnt find a way to do that differently.

Reply:
Oh, interesting, I missed that. Why are the dtypes diffferent? Is the regionmask adding some non-JSON-serializable type name somewhere?

Reply:
regionmask.region doesn’t have dtypes at least

Reply:
I don't have much experience with regionmask. If it isn't a container with some datatypes, would it be more appropriate to set `dtypes=None` for it?

Reply:
That is to say, if a `Schema` is intended to be a description of the data, then we should have this accurately reflect the shape and types of its contained data.

Reply:
Would be ok. But still the default get_schema wouldn’t retrieve dtypes=None?!

Reply:
I think it's fine to override `get_schema`, I didn't understand earlier that it was a different data model. Just wondering what the appropriate schema *should* be here.